src/tests/ftest/security/pool_security_acl.py

```diff
@@ -0,0 +1,403 @@
+        self.log.info("  (6)Verify pool read by: daos pool query --pool")
+        self.verify_pool_readwrite(svc, uuid, "read", expect=read)
+
+        # (7)Verify pool write operation
+        #    daos continer create --pool <uuid>
+        self.log.info("  (7)Verify pool write by: daos continer create --pool")
+        self.verify_pool_readwrite(svc, uuid, "write", expect=write)
+
+        # (8)Cleanup user and destroy pool
+        self.log.info("  (8)Cleanup user and destroy pool")
+        self.cleanup_user_group(num_user, num_group)
+        dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg.request.value = "pool"
+        dmg.action.value = "destroy --pool={}".format(uuid)
+        result = dmg.run()
+        return
+
+    def test_pool_acl_enforcement(self):
+        '''
+        Epic:
+            DAOS-1961: Testing related to DAOS security features.
+        Testcase description:
+            DAOS-2950: New pool create with pass-in ACL and connect
+                       credential verification.
+            DAOS-2952: Pool ACE/ACL permissions verification.
+            DAOS-2953: Pool ACL enforcement order verification.
+            DAOS-3611: Pool ACL verification after user removed from a
+                       granted group.
+            DAOS-3612: ACL update to remove/grant user/group access pool
+        Description:
+            Create pool with pass-in user and group acl permission,
+            verify pool user and group read, write, read-write and none
+            permissions enforcement with all forms of input under different
+            test sceanrios.
+        :avocado: tags=all,full_regression,security,pool_acl,sec_acl
+        '''
+        user_uid = os.geteuid()
+        user_gid = os.getegid()
+        current_user = pwd.getpwuid(user_uid)[0]
+        current_group = grp.getgrgid(user_gid)[0]
+        user_type = self.params.get("user", "/run/pool_acl/*")[0].lower()
+        permission, read, write = self.params.get("name", "/run/pool_acl/*")
+
+        user_types = ["owner", "user", "ownergroup", "group", "everyone"]
+        default_acl_entries = ["A::OWNER@:",
+                               acl_entry("user", current_user, ""),
+                               "A:G:GROUP@:",
+                               acl_entry("group", current_group, ""),
+                               "A::EVERYONE@:"]
+        test_acl_entries = ["", "", "", "", ""]
+
+        if permission.lower() == "none":
+            permission = ""
+        if permission not in PERMISSIONS:
+            self.fail("##permission {} is invalid, valid permissions are:"
+                      "'none', 'r', w', 'rw'".format(permission))
+        if user_type not in user_types:
+            self.fail("##user_type {} is invalid, valid user_types are: "
+                      "{}".format(user_type, user_types))
+        user_type_ind = user_types.index(user_type)
+        self.log.info("===>Start DAOS pool acl enforcement order Testcase: "
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments